libz-rs-sys/src/gz.rs

```diff
@@ -0,0 +1,230 @@
+pub use zlib_rs::c_api::*;
+
+use core::ffi::{c_char, c_int, c_uint, c_void, CStr };
+
+use std::ptr;
+use libc::{O_APPEND, O_CLOEXEC, O_CREAT, O_EXCL, O_LARGEFILE, O_RDONLY, O_TRUNC, O_WRONLY, SEEK_CUR, SEEK_END};
+use zlib_rs::deflate::Strategy;
+
+// For compatibility with the zlib C API, this structure exposes just enough of the
+// internal state of an open gzFile to support the gzgetc() C macro.
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct gzFile_s {
```

Comment:
Public items also need a doc comment (starting each line with `///`), that shows up in the documentation. It's fine to have some standard comments below that for internal details that are for the reader of the code, but not the user of the type. run `cargo doc --open` to see what the docs look like locally.